src/etc/inc/plugins.inc.d/core.inc

```diff
@@ -82,16 +82,48 @@ function core_services()
         $services[] = $pconfig;
     }
 
-    if (is_apinger_enabled()) {
-        $pconfig = array();
-        $pconfig['name'] = "apinger";
-        $pconfig['description'] = gettext("Gateway Monitoring Daemon");
-        $pconfig['php']['restart'] = array('setup_gateways_monitor');
-        $pconfig['php']['start'] = array('setup_gateways_monitor');
-        $pconfig['pidfile'] = '/var/run/apinger.pid';
-        $services[] = $pconfig;
-    }
+    if (!isset($config['system']['prefer_dpinger'])) {
+        if (is_apinger_enabled()) {
+            $pconfig = array();
+            $pconfig['name'] = "apinger";
+            $pconfig['description'] = gettext("Gateway Monitoring Daemon");
+            $pconfig['php']['restart'] = array('setup_gateways_monitor');
+            $pconfig['php']['start'] = array('setup_gateways_monitor');
+            $pconfig['pidfile'] = '/var/run/apinger.pid';
+            $services[] = $pconfig;
+        }
+    } else {
+        if (is_dpinger_enabled()) {
 
+        $pidfiles = glob("/var/run/dpinger_*.pid");
+
+        $gateways_arr_all = return_gateways_array(true);
+
+        foreach($gateways_arr_all as $name => $gateway) {
+            if (isset($gateway['monitor_disable'])) {
+                continue;
+            }
+
+            $pattern = "/^dpinger_({$name})~([^~]+)~([^~]+)\.pid$/";
+            foreach ($pidfiles as $pidfile) {
+                if (preg_match($pattern,basename($pidfile), $matches)) {
+                    $dpinger_pid = $pidfile;
+                }
+            }
+            $pconfig = array();
+            $pconfig['name'] = "dpinger {$gateway['name']}";
```

Comment:
and just 'dpinger' here

Reply:
No, there are multiple dpinger instances.